nav2_costmap_2d/include/nav2_costmap_2d/static_layer.hpp

```diff
@@ -56,51 +55,44 @@ class StaticLayer : public CostmapLayer
 public:
   StaticLayer();
   virtual ~StaticLayer();
+
   virtual void onInitialize();
   virtual void activate();
   virtual void deactivate();
   virtual void reset();
 
   virtual void updateBounds(
     double robot_x, double robot_y, double robot_yaw, double * min_x,
-    double * min_y,
-    double * max_x,
-    double * max_y);
+    double * min_y, double * max_x, double * max_y);
+
   virtual void updateCosts(
     nav2_costmap_2d::Costmap2D & master_grid,
     int min_i, int min_j, int max_i, int max_j);
 
   virtual void matchSize();
 
 private:
-  /**
-   * @brief  Callback to update the costmap's map from the map_server
-   * @param new_map The map to put into the costmap. The origin of the new
-   * map along with its size will determine what parts of the costmap's
-   * static map are overwritten.
-   */
-  void incomingMap(const nav_msgs::msg::OccupancyGrid::SharedPtr new_map);
-  void incomingUpdate(map_msgs::msg::OccupancyGridUpdate::ConstSharedPtr update);
-  void reconfigureCB();
-
+  void getParameters();
+  void getMap();
+  void processMap(const nav_msgs::msg::OccupancyGrid & new_map);
   unsigned char interpretValue(unsigned char value);
 
-  std::string global_frame_;  ///< @brief The global frame for the costmap
-  std::string map_frame_;  /// @brief frame that map is located in
-  bool subscribe_to_updates_;
-  bool map_received_;
-  bool has_updated_data_;
-  unsigned int x_, y_, width_, height_;
+  std::string global_frame_;
+  std::string map_frame_;
+
+  bool has_updated_data_{false};
+
+  unsigned int x_{0};
```

Comment:
Across this PR, I'm not sure why we're initializing variables at all, or doing so in the header file this way. Seems like it would make challenging to debug issues, or using this as a chance to set logical defaults.

Reply:
It was to ensure that all non-parameter member variables are initialized. I think it's a clean way to initialize members that don't appear in the constructor list. These will be initialized for any/all of the constructors.